homeassistant/components/screenlogic/binary_sensor.py

```diff
@@ -29,48 +30,56 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up entry."""
-    entities = []
-    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    entities: list[ScreenLogicBinarySensorEntity] = []
```

Comment:
You could write this something like the below to make it a bit more compact/readable
```python
    gateway_data = coordinator.gateway_data
    chemistry = gateway_data[SL_DATA.KEY_CHEMISTRY]
    config = gateway_data[SL_DATA.KEY_CONFIG]

    # Generic binary sensor
    entities.append(
        ScreenLogicStatusBinarySensor(coordinator, "chem_alarm", CODE.STATUS_CHANGED)
    )

    entities.extend(
        [
            ScreenlogicConfigBinarySensor(coordinator, cfg_sensor, CODE.STATUS_CHANGED)
            for cfg_sensor in config
            if cfg_sensor in SUPPORTED_CONFIG_BINARY_SENSORS
        ]
    )

    if config["equipment_flags"] & EQUIPMENT.FLAG_INTELLICHEM:
        # IntelliChem alarm sensors
        entities.extend(
            [
                ScreenlogicChemistryAlarmBinarySensor(
                    coordinator, chem_alarm, CODE.CHEMISTRY_CHANGED
                )
                for chem_alarm in chemistry[SL_DATA.KEY_ALERTS]
                if not chem_alarm.startswith("_")
            ]
        )

        # Intellichem notification sensors
        entities.extend(
            [
                ScreenlogicChemistryNotificationBinarySensor(
                    coordinator, chem_notif, CODE.CHEMISTRY_CHANGED
                )
                for chem_notif in chemistry[SL_DATA.KEY_NOTIFICATIONS]
                if not chem_notif.startswith("_")
            ]
        )

    if config["equipment_flags"] & EQUIPMENT.FLAG_CHLORINATOR:
        # SCG binary sensor
        entities.append(ScreenlogicSCGBinarySensor(coordinator, "scg_status"))
```